### PR TITLE
Miscellaneous

### PR DESCRIPTION
**Pull request by h-thurow**
Hello Brett, again a PR. Hope you will like it.

### CODE REVIEW

**Comment by h-thurow** [comment]
Both inserting objects with composite primary key and throwing an exception when multiple `@Id` annotations combined with `@GeneratedValue` annotion are encountered fails already in SansOrm 3.2.

**Comment by brettwooldridge** [comment]
Ah, that doesn’t surprise me too much.  In my company’s usage, I believe all @Id columns are simple auto generated INTEGER columns (one per table)...

**Comment by h-thurow** [comment]
SansOrm's test suite gets better and better :-)

File: src/main/java/com/zaxxer/sansorm/internal/OrmReader.java [hunk 1]
```diff
@@ -136,16 +136,14 @@ public static ResultSet statementToResultSet(final PreparedStatement stmt, final
...
    // COMPLEXITY:ON
 
-   public static <T> T statementToObject(final PreparedStatement stmt, final Class<T> clazz, final Object... args) throws SQLException
+   public static <T> T statementToObject(final PreparedStatement stmt, final T target, final Object... args) throws SQLException
    {
       populateStatementParameters(stmt, args);
 
       try (final ResultSet resultSet = stmt.executeQuery()) {
          if (resultSet.next()) {
-            final T target = clazz.newInstance();
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/sansorm/internal/OrmReader.java]: Removal of this code would be a breaking change for us.

I would suggest leaving the original method signature, but modifying the implementation to:
```Java
public static <T> T statementToObject(final PreparedStatement stmt, final Class<T> clazz, final Object... args) throws SQLException {
   final T target = clazz.newInstance();
   return this. statementToObject(stmt, target, args);
}
```

Comment (h-thurow) [reply src/main/java/com/zaxxer/sansorm/internal/OrmReader.java]: Sorry, you are right. I have missed that the method is public.

**Comment by h-thurow** [comment]
Thanks for the promotion 😃

**Comment by brettwooldridge** [comment]
Sure.  You're probably going to encounter use-cases that my code has not required, so I'm sure you may have more changes down the road.  The quality and correctness of your contributions as been very good, and is much appreciated.

**Pull request merged**